core/src/main/java/org/elasticsearch/transport/TcpTransport.java

```diff
@@ -768,19 +769,19 @@ protected final void doStop() {
         threadPool.generic().execute(() -> {
             globalLock.writeLock().lock();
             try {
-                for (Iterator<NodeChannels> it = connectedNodes.values().iterator(); it.hasNext(); ) {
-                    NodeChannels nodeChannels = it.next();
-                    it.remove();
-                    IOUtils.closeWhileHandlingException(nodeChannels);
-                }
-
                 for (Map.Entry<String, List<Channel>> entry : serverChannels.entrySet()) {
                     try {
                         closeChannels(entry.getValue());
                     } catch (Exception e) {
                         logger.debug("Error closing serverChannel for profile [{}]", e, entry.getKey());
                     }
                 }
+
+                for (Iterator<NodeChannels> it = connectedNodes.values().iterator(); it.hasNext(); ) {
```

Comment:
> why did you move this?

The first thing I wanted to do is stopping this transport from accepting any incoming connections and only when we're done with that I wanted to close connected nodes. This change is not really necessary but to me this is the cleaner option. I can add a comment.

> Did you find out why it needs to be done twice? if so, please add a comment :)

It's not needed: We're operating under a lock (which is also used by `#connectToNode()`(!)) and we've just capped all incoming and outgoing connections. It was just a leftover and I'll remove it in my next commit.